app/src/main/java/com/google/android/ground/ui/home/mapcontainer/jobs/JobMapAdapter.kt

```diff
@@ -0,0 +1,398 @@
+import androidx.compose.material.icons.filled.Clear
+import androidx.compose.material3.BottomSheetDefaults
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.pluralStringResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.google.android.ground.R
+import com.google.android.ground.databinding.BasemapLayoutBinding
+import com.google.android.ground.databinding.MenuButtonBinding
+import com.google.android.ground.model.job.getDefaultColor
+import com.google.android.ground.model.locationofinterest.LocationOfInterest
+import com.google.android.ground.ui.common.LocationOfInterestHelper
+import com.google.android.ground.ui.theme.AppTheme
+import kotlinx.coroutines.launch
+
+/** Manages a set of [Composable] components that renders [LocationOfInterest] cards and dialogs. */
+class JobMapAdapter(private val getSubmissionCount: suspend (loi: LocationOfInterest) -> Int) {
+  lateinit var basemapLayoutBinding: BasemapLayoutBinding
+  lateinit var menuBinding: MenuButtonBinding
+  private var collectDataListener: MutableState<(DataCollectionEntryPointData) -> Unit> =
+    mutableStateOf({})
+  private var canUserSubmitData = mutableStateOf(false)
+  private var activeLoi: MutableState<DataCollectionEntryPointData.SelectedLoiSheetData?> =
+    mutableStateOf(null)
+  private val newLoiJobs: MutableList<DataCollectionEntryPointData.AdHocDataCollectionButtonData> =
+    mutableStateListOf()
+  private var selectedFeatureListener: ((String?) -> Unit) = {}
+  private val jobModalOpened = mutableStateOf(false)
+  private val jobCardOpened = mutableStateOf(false)
+  private val submissionCount = mutableIntStateOf(-1)
+
+  fun render() {
```

Comment:
Can we make `basemapLayoutBinding.bottomContainer` a parameter of this method?

Reply:
Yes! I was able to pass through the bindings so we don't have any loose references. Done.

Reply:
Thanks